pkg/authorization/apibinding_authorizer.go

```diff
@@ -235,12 +215,11 @@ func (a *apiBindingAccessAuthorizer) getAPIBindingReference(attr authorizer.Attr
 	return nil, false, nil
 }
 
-func (a *apiBindingAccessAuthorizer) getAPIExport(exportRef *apisv1alpha1.ExportReference) (*apisv1alpha1.APIExport, bool, error) {
-	objs, err := a.apiExportIndexer.ByIndex(byWorkspaceIndex, exportRef.Workspace.Path)
+func GetAPIExportByReference(apiExportIndexer cache.Indexer, exportRef *apisv1alpha1.ExportReference) (*apisv1alpha1.APIExport, bool, error) {
```

Comment:
Why are these two not simply indices? As opposed to doing the broader cluster index and iterating?

Reply:
That was in the original code, I reused it. However, we could build a dedicated index that uses api export's cluster name + api export's bound resource group + export's bound resource.

Are compound index keys idiomatic?

Reply:
@sttts ^ ?

Reply:
Use type-safe index key formatters to ensure it is used correctly, see https://github.com/kcp-dev/apimachinery/blob/main/pkg/cache/controller.go

Reply:
We have a handful of exports per workspace. Iterating is not an issue here. If we had a hundred or a thousand of those objects we would care.